Clarify render status UI

imsyphia opened the pull request:
The current render status pane is somewhat messy with odd and inconsistent formatting, leading to issues like #1213, where `Rendering: 5 of 7` can be misinterpreted as _currently_ rendering 5 of 7. I tried to make it overall more readable and intuitively organized.

This (mostly) fixes #1213. It can still be misinterpreted, but it should be somewhat clearer without having to add additional visual clutter like "5 of 7 done".

I've attached before/after images below for comparison.

![chunkyuibefore](https://github.com/chunky-dev/chunky/assets/85633285/33b31e7a-6177-4fd3-9213-e9c851cc15e8)

![chunkyuiafter](https://github.com/chunky-dev/chunky/assets/85633285/19dc4bac-4f11-4008-a704-abec80bbe91f)

jackjt8 commented:
This is definitely an improvement. Having Render time and ETA on the same line in the same format is really nice.

I am still not 100% happy with the "Rendering" line... but as you have said.. I'm not sure how we can "fix" it without adding more visual clutter. I wonder if hover text with more verbose information would be a good or bad idea.. It's clutter.. just not visible? Or would a button to toggle verbose status UI be silly?

leMaik commented:
Great first PR! 🎉 

How about calculating a percentage? Eg. "Rendering (42%)", if that number is not zero? The numbers are spp for rendering but otherwise pretty arbitrary anyways, and a percentage is nicer in formatting imho.

imsyphia commented:
> Great first PR! 🎉
> 
> How about calculating a percentage? Eg. "Rendering (42%)", if that "of" number is not zero? The numbers are spp for rendering but otherwise pretty arbitrary anyways, and a percentage is nicer in formatting imho.

I actually did try adding a percentage, but the combination of counter and percentage looked a little clunky. Doing `Rendering | 42%` or something similar could totally work, though. And it would fix the "Rendering" issue as @jackjt8 stated. I'll try it when I get around to it.

My only concern is that having a specific number could be useful for debugging in certain circumstances, so it may be worth adding that information elsewhere if it's not already available.

imsyphia commented:
Also, this may fall slightly outside the scope of this PR, but job names are inconsistent. Some are nouns, like `Preview`, and others are active verbs, like `Rendering`. If this is a simple fix I'll likely include it, since it is mostly relevant to the status pane, but if it's more complicated I'll make a separate issue.

leMaik commented:
Feel free to update the task names to verbs, while you're at it. 👍

imsyphia commented:
I've replaced the `N of M` indicator with percentage progress, and renamed "Render time" to "Time", since it should be clear enough from context, and is not necessarily restricted to rendering.

The percentage indicator is only two digits because additional digits on the percentage progress looked awkward and was unnecessary given that there's already a timer, progress bar, and SPP counter to indicate that progress is being made.

The "Preview" task is now named "Displaying preview", since it was the only inconsistent one. I avoided "Rendering preview" because a newer user could confuse it with the actual rendering process. 

![chunkyuiafter2](https://github.com/chunky-dev/chunky/assets/85633285/86260fd7-ec4d-499d-8188-548fe38e81f2)

imsyphia commented:
This is definitely outside the scope of this PR, but I was thinking that the top-right of the pane doesn't need to be exclusively SPP and SPS. It could be used as a general-purpose info/status list for the current task. Should I make an issue for that?

leMaik commented on chunky/src/java/se/llbit/chunky/renderer/PreviewRenderer.java:
```diff
@@ -54,7 +54,7 @@ public String getDescription() {
   public void render(DefaultRenderManager manager) throws InterruptedException {
     TaskTracker.Task task = manager.getRenderTask();
-    task.update("Preview", 2, 0, "");
+    task.update("Displaying preview", 2, 0, "");
```
It's not the displaying that takes a bit of time, it's rendering the preview:

```suggestion
    task.update("Rendering preview", 2, 0, "");
```

imsyphia replied:
I initially chose "Rendering preview" as well, but I chose to use "Displaying preview" to avoid confusion with the actual rendering process, especially for new users. Maybe a different wording could be used?

leMaik replied:
Preparing preview?

leMaik commented on chunky/src/java/se/llbit/chunky/ui/controller/ChunkyFxController.java:
```diff
@@ -162,17 +162,23 @@ public class ChunkyFxController
-        progressLbl.setText(String.format("%s: %s of %s", task, decimalFormat.format(done),
-            decimalFormat.format(target)));
+        if (target - start > 0) {
+          progressLbl.setText(String.format("%s - %s%%", task, Math.round(100 * (double) done / (target - start))));
```
`%s` is for strings, maybe we can even have one decimal digit (when progress is slow):

```suggestion
          progressLbl.setText(String.format("%s - %.1f%%", task, Math.round(100 * (double) done / (target - start))));
```

imsyphia replied:
Yeah, `%f` makes more sense than `%s`.

Personally, I found that adding a decimal digit gave a bit too much visual weight to the element. Also, the progress bar and SPP counter should already give some indication of progress. I'll experiment with it again, though.

imsyphia replied:
Adding replacing the hyphen with an en dash fixes the visual weight issue, so I've gone ahead and added the extra digit.

leMaik commented on chunky/src/java/se/llbit/chunky/ui/controller/ChunkyFxController.java:
```diff
@@ -162,17 +162,23 @@ public class ChunkyFxController
-        progressLbl.setText(String.format("%s: %s of %s", task, decimalFormat.format(done),
-            decimalFormat.format(target)));
+        if (target - start > 0) {
+          progressLbl.setText(String.format("%s - %s%%", task, Math.round(100 * (double) done / (target - start))));
```
```suggestion
          progressLbl.setText(String.format("%s - %.1f%%", task, Math.round(100 * (double) done / (target - start))));
```

leMaik commented on chunky/src/java/se/llbit/chunky/ui/controller/ChunkyFxController.java:
```diff
@@ -205,8 +211,7 @@ public GUIRenderListener(ChunkyFxController renderControls) {
         int hours = (int) (time / 3600000);
-        gui.renderTimeLbl.setText(String
-            .format("Render time: %d hours, %d minutes, %d seconds", hours, minutes, seconds));
+        gui.renderTimeLbl.setText(String.format("Time: %d:%02d:%02d", hours, minutes, seconds));
```
Geat! 👏

leMaik reviewed (changes requested):
Great first PR 🎉
I just have some minor wording and percentage formatting suggestions. Overall, this is fine and the changes look great.

leMaik commented:
> My only concern is that having a specific number could be useful for debugging in certain circumstances, so it may be worth adding that information elsewhere if it's not already available.

Regarding this concern: Maybe have the full numbers as tooltip? Or maybe after the percentage, like `Rendering - 20% (200 of 1000)`?

imsyphia commented:
Here's what it now looks like, just for reference.

![chunkyuiafter3](https://github.com/chunky-dev/chunky/assets/85633285/0f974d07-d11a-4be6-8569-3a5d7d6d6707)
